W-17831880- Document conversion of keystores to BCFKS (#2930)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/fips-140-2-compliance-support.adoc

* TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
* TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA

== FIPS 140-2 Compliant Keystore Formats

> KeyStores or TrustStores in Mule apps are typically formatted as `PKCS12` or `JKS`.

Reviewer: ```suggestion
Keystores or truststores in Mule apps are typically formatted as `PKCS12` or `JKS`.
```

> These formats aren't FIPS compliant and must be converted to `BCFKS` format for compliance.

Reviewer 2: ```suggestion
These formats aren't FIPS compliant. For compliance, convert them to `BCFKS` format.
```

> To convert a `JKS` keystore to `BCFKS` format, ensure that you: 

Reviewer 2: ```suggestion
To convert a `JKS` keystore to `BCFKS` format, make sure that you: 
```

> . Use Java 8 or later and `JAVA_HOME` is set.

Reviewer 2: ```suggestion
. Use Java 8 or later and make sure that `JAVA_HOME` is set.
```

> . Verify that you are using Java 8 or later and `JAVA_HOME` is set.

Reviewer: ```suggestion
. Use Java 8 or later and `JAVA_HOME` is set.
```

. Download the `bc-fips.jar` file from the https://www.bouncycastle.org/download/bouncy-castle-java-fips/[Bouncy Castle website].

> . Use the following example command to convert a keystore to `BCFKS` format:

Reviewer 2: ```suggestion
. Use this example command to convert a keystore to `BCFKS` format:
```

----
BC_FIPS_JAR=${BC_PATH}/bc-fips-*.jar  # Replace with a correct path
OLD_KEYSTORE="keystore.jks"           # Replace with the keystore to convert
OLD_PASSWD="changeit"                 # Replace with the keystore password
NEW_KEYSTORE="keystore.bcfks"         # Replace with the new keystore
NEW_PASSWD="changeit"                 # Replace with the new keystore password

keytool -importkeystore \
 -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
 -providerpath ${BC_FIPS_JAR} \
 -srckeystore  ${OLD_KEYSTORE}   -srcstoretype JKS       -srcstorepass ${OLD_PASSWD} \
 -destkeystore ${NEW_KEYSTORE}   -deststoretype BCFKS    -deststorepass ${NEW_PASSWD}
----

[IMPORTANT]

> If you are using Java 17 or later, add the `-J--add-exports=java.base/sun.security.provider=ALL-UNNAMED` parameter to the keytool command before the `-importkeystore` parameter.

Reviewer 2: ```suggestion
If you're using Java 17 or later, add the `-J--add-exports=java.base/sun.security.provider=ALL-UNNAMED` parameter to the keytool command before the `-importkeystore` parameter.
```

[start=4]

> . Update the xref:tls-configuration.adoc[TLS configuration] in the Mule configuration file to use the new keystore or truststore:

Reviewer 2: ```suggestion
. To use the new keystore or truststore, update the xref:tls-configuration.adoc[TLS configuration] in the Mule configuration file:
```

> . Update the xref:tls-configuration.adoc[TLS configuration] in the Mule configuration file to use the new KeyStore or TrustStore:

Reviewer: ```suggestion
. Update the xref:tls-configuration.adoc[TLS configuration] in the Mule configuration file to use the new keystore or truststore:
```
